### DISCUSSION TITLE
Support for Logical OR Operator in Assert Section

### DISCUSSION THREAD

**Discussion by fml09** [Orange-OpenSource/hurl#2600] [Q&A]
Hey there! I have a situation where I need to make assertions based on two possible responses from the server. Essentially, I want to include the logical "OR" operator in the Assert section. For example, I want the test to pass if the HTTP Status code is between 200 and 300, 'or' if the Status code is 400 and the Body contains a specific text. Here's what I have in mind:
```
GET https://example.com/

HTTP *
[Asserts]
status >= 200
status < 300

OR
[Asserts]
status == 400
body contains "TEXT"
```

**Comment by fabricereix** [comment]
Hi @fml09,
Adding an `OR` operator would be indeed great, but this is a major change in term of syntax.
We need to think about it.

In a near future, we could already add the `oneOf` predicate.
```
[Asserts]
status oneOf [200, 400]
```

**Reply by iwinux** [reply]
Any workaround before this gets implemented?

**Reply by jcamiel** [reply]
Hi @iwinux 

You can play with [filters](https://hurl.dev), for instance `toString` can convert number to string and apply a `matches` predicate:

```hurl
GET https://example.com
HTTP *
[Asserts]
# Test status 201 or 202 or 203 or 204
status toString matches /20[1-4]/ 
```
